app/packs/src/apps/mydb/elements/details/samples/SampleDetails.js

```diff
@@ -1404,39 +1423,56 @@ export default class SampleDetails extends React.Component {
   }
 
   svgOrLoading(sample) {
+    const { molfileConverstionRequired } = this.state;
     let svgPath = '';
     if (this.state.loadingMolecule) {
       svgPath = '/images/wild_card/loading-bubbles.svg';
     } else {
       svgPath = sample.svgPath;
     }
     const className = svgPath ? 'svg-container' : 'svg-container-empty';
+
     return (
       sample.can_update
         ? (
-          <>
+          <div
+            className={className}
+            style={{ position: 'relative' }}
+
+          >
             <div
-              className={className}
-              style={{ position: 'relative' }}
               onClick={this.showStructureEditor.bind(this)}
-              onKeyPress={this.showStructureEditor.bind(this)}
+              onKeyPress
               role="button"
               tabIndex="0"
             >
               <Glyphicon className="pull-right" glyph="pencil" />
               <SVG key={svgPath} src={svgPath} className="molecule-mid" />
             </div>
+
+            {molfileConverstionRequired &&
+              <div
+                className={"file-type-conversion-button"}
+                onClick={this.convertFileContentWithIndigo}
+                onKeyPress
+                role="button"
+                tabIndex="1"
+              >
+                <Glyphicon className="pull-right" glyph="refresh" />
+              </div>
+            }
             <MolViewerBtn
-              className="structure-editor-container"
+              className={["structure-editor-container", !molfileConverstionRequired && "structure-editor-container-without-convert-button"].join(" ")}
```

Comment:
better to use classnames Library:
  className={classNames('structure-editor-container', {
    'structure-editor-container-without-convert-button': !molfileConversionRequired
  })}
  or template Literals

or